plugin/trino-resource-group-managers/src/main/java/io/trino/plugin/resourcegroups/db/ResourceGroupGlobalPropertiesReducer.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.resourcegroups.db;
+
+import io.airlift.units.Duration;
+import org.jdbi.v3.core.result.RowReducer;
+import org.jdbi.v3.core.result.RowView;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class ResourceGroupGlobalPropertiesReducer
+        implements RowReducer<ResourceGroupGlobalProperties.Builder, ResourceGroupGlobalProperties>
+{
+    @Override
+    public ResourceGroupGlobalProperties.Builder container()
+    {
+        return ResourceGroupGlobalProperties.builder();
+    }
+
+    @Override
+    public void accumulate(ResourceGroupGlobalProperties.Builder container, RowView rowView)
+    {
+        String name = rowView.getColumn("name", String.class);
+        Optional<Duration> value = Optional.ofNullable(rowView.getColumn("value", String.class)).map(Duration::valueOf);
+
+        switch (name) {
+            case "cpu_quota_period":
+                container.setCpuQuotaPeriod(value);
+                break;
+            case "physical_data_scan_quota_period":
+                container.setPhysicalDataScanQuotaPeriod(value);
+                break;
+            // Unrecognized properties are simply ignored
```

Comment:
Do you expect other properties to be there? If not, could we throw an exception here?

Reply:
Yup, that's a good idea. No other properties should be there after this [change](https://github.com/trinodb/trino/pull/25113#discussion_r2231517532)

Added that and squashed commits. @wendigo